lib/posix/pthread_mutex.c

```diff
@@ -9,6 +9,18 @@
 #include <wait_q.h>
 #include <posix/pthread.h>
 
+/* Memory pool for pthread space */
+K_MEM_POOL_DEFINE(posix_mutex_sem, sizeof(struct k_sem),
+		  sizeof(struct k_sem), 32, 4);
+
+/*
+ *  Default mutex attrs.
+ */
+static pthread_mutexattr_t def_attr = {
+	PTHREAD_MUTEX_DEFAULT, /* type */
```

Comment:
Please define this as `static const` and use named initializers: `{ .type = PTHREAD_MUTEX_DEFAULT, .protocol = PTHREAD_PRIO_NONE }`.